src/Discord.Net.Commands/Attributes/Preconditions/RequireOwnerAttribute.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Discord;
+
+namespace Discord.Commands
+{
+    /// <summary>
+    /// Require that the command is invoked by the owner of the bot.
+    /// </summary>
+    /// <remarks>This precondition will only work if the bot is a bot account.</remarks>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class RequireOwnerAttribute : PreconditionAttribute
+    {
+        private IApplication application;
+
+        public override async Task<PreconditionResult> CheckPermissions(CommandContext context, CommandInfo command, IDependencyMap map)
+        {
+            if (application == null)
+                application = await context.Client.GetApplicationInfoAsync();
```

Comment:
The precondition instance is cached in the CommandInfo/ModuleInfo type - so this should not be an issue as long as the command isn't loaded/unloaded multiple times.

However, I think that caching ApplicationInfo on the DiscordSocketClient would be a good idea for the lib overall - I would use it in my preconditions to prevent ratelimits from occuring.